### PR TITLE
feat(http-client-csharp): bump to tcgc 0.47.4

### LINKED ISSUE #4938: feat(http-client-csharp): bump tcgc 0.47.4
## Description

Upgrade TCGC to 0.47.4. There are a few non-trivial changes in this future. So, we'd better upgrade it separately first to avoid huge change when upgrading to next version of typespec.

### PR DESCRIPTION
**Pull request by archerzz**
- upgrade tcgc to 0.47.4
- upgrade cadl-ranch to latest version

resolve #4938

### CODE REVIEW

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/array/tspCodeModel.json [hunk 1]
```diff
@@ -2174,15 +2174,7 @@
...
-         "$id": "223",
-         "kind": "nullable",
-         "type": {
-          "$id": "224",
-          "kind": "float32",
-          "name": "float32",
-          "crossLanguageDefinitionId": "TypeSpec.float32",
-          "decorators": []
-         }
+         "$ref": "215"
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/array/tspCodeModel.json:2177]: type id `215` is the same type:
```json
"valueType": {
  "$id": "215",
  "kind": "nullable",
  "type": {
  "$id": "216",
  "kind": "float32",
  "name": "float32",
  "crossLanguageDefinitionId": "TypeSpec.float32",
  "decorators": []
}
```

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/dictionary/tspCodeModel.json [hunk 2]
```diff
@@ -2492,15 +2492,7 @@
...
-         "$id": "265",
-         "kind": "nullable",
-         "type": {
-          "$id": "266",
-          "kind": "float32",
-          "name": "float32",
-          "crossLanguageDefinitionId": "TypeSpec.float32",
-          "decorators": []
-         }
+         "$ref": "256"
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/dictionary/tspCodeModel.json:2495]: same

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/union/tspCodeModel.json [hunk 3]
```diff
@@ -566,42 +566,7 @@
...
-          {
-           "$id": "71",
-           "kind": "boolean",
-           "name": "boolean",
-           "crossLanguageDefinitionId": "TypeSpec.boolean",
-           "decorators": []
-          }
-         ],
-         "decorators": []
+         "$ref": "58"
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/union/tspCodeModel.json:569]: type id `58` is actually the same:
```json
{
  "$id": "58",
  "kind": "union",
  "name": "MixedTypesCasesModel",
  "variantTypes": [
    {
      "$id": "59", // note it's referenced in the deleted definition
      "kind": "model",
      "name": "Cat",
      "crossLanguageDefinitionId": "Type.Union.Cat",
      "usage": "Input,Output,Json",
      "decorators": [],
      "properties": [
        {
          "$id": "60",
          "kind": "property",
          "name": "name",
          "serializedName": "name",
          "type": {
            "$id": "61",
            "kind": "string",
            "name": "string",
            "crossLanguageDefinitionId": "TypeSpec.string",
            "decorators": []
          },
          "optional": false,
          "readOnly": false,
          "discriminator": false,
          "flatten": false,
          "decorators": [],
          "crossLanguageDefinitionId": "Type.Union.Cat.name"
        }
      ]
    },
    {
      "$id": "62",
      "kind": "constant",
      "valueType": {
        "$id": "63",
        "kind": "string",
        "name": "string",
        "crossLanguageDefinitionId": "TypeSpec.string",
        "decorators": []
      },
      "value": "a",
      "decorators": []
    },
    {
      "$id": "64",
      "kind": "int32",
      "name": "int32",
      "crossLanguageDefinitionId": "TypeSpec.int32",
      "decorators": []
    },
    {
      "$id": "65",
      "kind": "boolean",
      "name": "boolean",
      "crossLanguageDefinitionId": "TypeSpec.boolean",
      "decorators": []
    }
  ],
  "decorators": []
}
```

**Comment by archerzz** [comment]
double confirmed that tcgc `0.47.4` fixes duplicated nullable types and union types.

**Review by JoshLove-msft — APPROVED** [review]

**Comment by archerzz** [comment]
This PR is blocked by problematic typespec definition of `Language.Conversation` service. I've notified the service team to fix the problem

**Comment by archerzz** [comment]
PR to fix the spec: https://github.com/Azure/azure-rest-api-specs/pull/31416

**Pull request merged**